src/dataworker/Dataworker.ts

```diff
@@ -13,40 +21,134 @@ export class Dataworker {
+
+        // Store deposits whose destination chain is the selected chain as unfilled deposits and set the initial fill
+        // amount remaining equal to the full deposit amount minus any valid fill amounts.
+        // Remove any deposits that have no unfilled amount (i.e that have an unfilled amount of 0) and append the
+        // remaining deposits to the unfilledDeposits array.
+        const depositsForDestinationChain = originClient.getDepositsForDestinationChain(destinationChainId);
+        this.logger.debug({
+          at: "Dataworker",
+          message: `Found ${depositsForDestinationChain.length} deposits for destination chain ${destinationChainId}`,
+          originChainId,
+          destinationChainId,
+        });
+
+        const unfilledDepositsForDestinationChain = depositsForDestinationChain
+          .map((deposit) => {
+            return { data: deposit, unfilledAmount: destinationClient.getValidUnfilledAmountForDeposit(deposit) };
+          })
+          .filter((deposit) => deposit.unfilledAmount.gt(0));
+
+        if (unfilledDepositsForDestinationChain.length > 0) {
+          assign(unfilledDeposits, [destinationChainId], unfilledDepositsForDestinationChain);
+        } else {
+          this.logger.debug({
+            at: "Dataworker",
+            message: `All deposits are filled`,
+            originChainId,
+            destinationChainId,
+          });
+        }
+
+        // Grab all valid fills submitted to the destination spoke pool.
+        const fillsOnDestinationChain = destinationClient.getFills();
+        const validFillsOnDestinationChain = fillsOnDestinationChain.filter((fill) => {
+          // For each fill, see if we can find a deposit sent from the origin client that matches it.
+          for (const deposit of depositsForDestinationChain) {
+            // @dev: It doesn't matter which client we call validateFillForDeposit() on as the logic is
+            // chain agnostic.
+            // @dev: All of the deposits returned by `getDepositsForDestinationChain` will include the expected realized
+            // lp fee % for the deposit quote time. If this fill does not have the same realized lp fee %, then it will
+            // be ignored.
+
+            if (destinationClient.validateFillForDeposit(fill, deposit)) return true;
+            else continue;
+          }
+
+          // No deposit matched, this fill is invalid.
+          return false;
+        });
+        this.logger.debug({
+          at: "Dataworker",
+          message: `Found ${validFillsOnDestinationChain.length} fills on destination ${destinationChainId} matching origin ${originChainId}`,
+          originChainId,
+          destinationChainId,
+        });
+
+        for (const fill of validFillsOnDestinationChain) {
+          assign(fillsToRefund, [fill.repaymentChainId, fill.relayer], [fill]);
+        }
```

Comment:
might need to be listed. not sure if this is too long.
```suggestion
        for (const fill of validFillsOnDestinationChain)assign(fillsToRefund, [fill.repaymentChainId, fill.relayer], [fill]);
```

Reply:
would a `forEach` make you happy?

Reply:
what you have is fine. just you dont need the extra pair of `{` & `}` :)

Reply:
gotcha i ended up using `forEach`